### PR TITLE
Fix image size and use Img tag instead of background url

### PR DESCRIPTION
**Pull request by rerissondaniel**
#### What is the purpose of this pull request?

Use resized images in product-summary instead of the original image.

#### What problem is this solving?

Even though the image needed in this component was small, it always loaded the maximum size of the image. Now it loads with 500px, also making it possible to customize this value through props.

#### How should this be manually tested?
[workspace](https://imageresizeps--storecomponents.myvtex.com/)

#### Types of changes
- [X] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: react/components/Image.js [hunk 1]
```diff
@@ -4,18 +4,46 @@ import PropTypes from 'prop-types'
 import productSummary from '../productSummary.css'
 
 /** Image component with 1:1 aspect ratio */
-export default function Image({ src, className }) {
+export default function Image({ src, className, width, height }) {
+  const resizeSrc = src => {
```
Comment (luciannojunior) [inline react/components/Image.js]: Couldn't we reuse [this](https://github.com/vtex-apps/product-details/blob/master/react/utils/generateUrl.js)?

File: react/components/Image.js [hunk 2]
```diff
@@ -4,18 +4,46 @@ import PropTypes from 'prop-types'
...
+    //Check of src is already in a size defined by the client
+    const isWithMeasures = new RegExp(/[1-9]+-(:?([1-9]+)|(auto))/);
+    if (isWithMeasures.exec(src)) {
+      return src
+    }
+
+    const imageId = src.slice(src.indexOf("/ids/")).split("/")[2]
```
Comment (luciannojunior) [inline react/components/Image.js]: The `[2]` accessor is error prone. Use Ramda's `path` instead

File: react/components/Image.js [hunk 3]
```diff
@@ -4,18 +4,46 @@ import PropTypes from 'prop-types'
...
+    const withSuffix = `${imageId}-${width}-${height}`
+
+    return src.replace(imageId, withSuffix)
+  }
+
   return (
     <div
       className={`${className} ${productSummary.aspectRatio} w-100`}
-      style={{ backgroundImage: `url(${src})` }}
+      style={{ backgroundImage: `url(${resizeSrc(src)})` }}
```
Comment (luciannojunior) [inline react/components/Image.js]: The PR title says that now the `Img` tag will be used. Wouldn't it change something here?

Comment (rerissondaniel) [reply react/components/Image.js]: It also says WIP. Sorry for having posted so early in our channel

File: react/components/Image.js (hunk 1 above)
Comment (rerissondaniel) [reply react/components/Image.js]: We could, but how?

Comment (luciannojunior) [reply react/components/Image.js]: You could:

1) import the function exported from `product-details`'s utils.
2) create a common component on `store-components` with the algorithm.

I will check with @brunojdo

Comment (rerissondaniel) [reply react/components/Image.js]: I think that inserting a dependency of `product-details` only for that is too costly

File: react/index.js [hunk 4]
```diff
@@ -114,7 +122,7 @@ class ProductSummary extends Component {
     } = product
 
     let img = (
-      <Image className={`${productSummary.image}`} alt={name} src={imageUrl} />
+      <Image className={`${productSummary.image}`} alt={name} src={imageUrl} width={imageWidth} height={imageHeight}/>
```
Comment (brunojdo) [inline react/index.js]: It's not the productName I think that is the skuName

**Comment by rerissondaniel** [comment]
> Fix tests

There's a lot of configuration to the tests to work, since that is done in [this PR](https://github.com/vtex-apps/product-summary/pull/84) I'm gonna wait for it to be accepted

File: react/components/Image.js (hunk 1 above)
Comment (gustavorosolem) [reply react/components/Image.js]: This same function is being used on **minicart** too.
https://github.com/vtex-apps/minicart/blob/master/react/utils/urlHelpers.js
This function should be a lib or something that allow using the same code.

File: react/components/Image.js [hunk 5]
```diff
@@ -1,21 +1,49 @@
...
+export default function Image({ src, className, width, height, alt }) {
+  const resizeSrc = src => {
+    //Check of src is already in a size defined by the client
+    const isWithMeasures = new RegExp(/[1-9]+-(:?([1-9]+)|(auto))/);
+    if (isWithMeasures.exec(src)) {
+      return src
+    }
+
+    const imageId = path('2', src.slice(src.indexOf("/ids/")).split("/"))
+    const withSuffix = `${imageId}-${width}-${height}`
```
Comment (gustavorosolem) [inline react/components/Image.js]: Return the query string used on the new **vtex.file-manager** too.
Example: https://github.com/vtex-apps/product-details/blob/master/react/utils/generateUrl.js#L16

**Pull request closed without merging**